packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -430,3 +463,153 @@ class _EditableLineWidget extends LeafRenderObjectWidget {
     return new TextSpan(style: style, text: text);
   }
 }
+
+enum _TextSelectionHandleSide { left, right }
+
+class _TextSelectionHandleOverlay extends StatefulWidget {
+  _TextSelectionHandleOverlay({
+    Key key,
+    this.onSelectionHandleChanged,
+    this.selection,
+    this.renderObject
+  }) : super(key: key);
+
+  final TextSelection selection;
+  final ValueChanged<TextSelection> onSelectionHandleChanged;
+  // TODO(mpcomplete): what if the renderObject is removed or replaced, or
+  // moves? Not sure what cases I need to handle, or how to handle them.
+  final RenderEditableLine renderObject;
+
+  @override
+  _TextSelectionHandleOverlayState createState() => new _TextSelectionHandleOverlayState();
+}
+
+/// The state for an overlay that displays 2 selection handles. This is intended
+/// to be used only when there is text selected (e.g. the TextSelection is not
+/// collapsed).
+class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay> {
+  TextSelection selection;
+  List<Point> endpoints;
+
+  @override
+  void initState() {
+    super.initState();
+    selection = config.selection;
+    endpoints = config.renderObject.getEndpointsForSelection(selection);
+  }
+
+  @override
+  void didUpdateConfig(_TextSelectionHandleOverlay oldConfig) {
+    setState(() {
+      selection = config.selection;
+      endpoints = config.renderObject.getEndpointsForSelection(selection);
+    });
+  }
+
+  Point _dragPosition;
+  void _dragStart(Point position, _TextSelectionHandleSide side) {
+    _dragPosition = position;
+  }
+
+  void _dragUpdate(double delta, _TextSelectionHandleSide side) {
+    _dragPosition = _dragPosition + new Offset(delta, 0.0);
+    TextPosition position = config.renderObject.getPositionForPoint(_dragPosition);
+
+    TextSelection newSelection;
+    if (side == _TextSelectionHandleSide.left) {
+      newSelection = new TextSelection(
+        baseOffset: position.offset,
+        extentOffset: selection.extentOffset
+      );
+    } else {
+      newSelection = new TextSelection(
+        baseOffset: selection.baseOffset,
+        extentOffset: position.offset
+      );
+    }
+
+    if (newSelection.baseOffset >= newSelection.extentOffset)
+      return;
+
+    setState(() {
+      selection = newSelection;
+      endpoints = config.renderObject.getEndpointsForSelection(selection);
```

Comment:
This isn't correct.  You're just going to blow way these fields when your parent rebuilds anyway.  Instead, you should just call onSelectionHandleChanged and rely on your parent rebuilding your with the new values.